book/src/plumbing.md

```diff
@@ -0,0 +1,279 @@
+# Plumbing
+
+This chapter documents the code that salsa generates and its "inner workings".
+We refer to this as the "plumbing".
```

Comment:
Such internals docs tend to go obsolete, so it might be a good idea to add "as of 2020-06-24"

Reply:
+1, although I think it also falls on us to ensure that any major edits to the plumbing edit this book in the same PR.